internal/handlers/common/aggregations/stages/projection/projection.go

```diff
@@ -62,21 +63,17 @@ func ValidateProjection(projection *types.Document) (*types.Document, bool, erro
 			return nil, false, lazyerrors.Error(err)
 		}
 
-		if strings.HasPrefix(key, "$") {
-			return nil, false, commonerrors.NewCommandErrorMsg(
-				commonerrors.ErrNotImplemented,
-				fmt.Sprintf("projection operator $ is not supported in %s", key),
-			)
-		}
-
 		var result bool
 
 		switch value := value.(type) {
 		case *types.Document:
-			return nil, false, commonerrors.NewCommandErrorMsg(
-				commonerrors.ErrNotImplemented,
-				fmt.Sprintf("projection expression %s is not supported", types.FormatAnyValue(value)),
-			)
+			_, err := operators.Get(value, "$project", key)
+			if err != nil {
+				return nil, false, err
+			}
+
+			validated.Set(key, value)
```

Comment:
I see now projection supports an operator, but we don't yet have an implementation of any operator yet. So code isn't tested. Could you add todo for testing? I assume this will be tested in https://github.com/FerretDB/FerretDB/issues/2678

By the way, did you test this manually?

Reply:
Thanks, the error is now returned slightly later in the code :+1: